### DISCUSSION TITLE
Node Firebase IN

### DISCUSSION THREAD

**Discussion by Knabbler** [GogoVega/node-red-contrib-firebase-realtime-database#30] [Ideas]
Dear Gauthier,

as far as I understood, the node is used as an event listener. 
Playing around with it I notice, that it fires also whenever a flow is deployed. 
My question is, whether this is intentionally or should better be suppressed?

Kind regards

Knabbler

**Comment by GogoVega** [comment]
Hi Knabbler,

The `Firebase in` node is an event listener, which means you specify the type of event and you will receive a payload related to this event.

*Why do I receive a message when I deploy my flow?*
**You receive a payload when the listener is instantiate**, it works like "get last state on startup".

To avoid receiving a payload every time you deploy, I suggest you changing the deployment of Node-RED to `Modified Nodes`. Indeed the first two options acts as a restart of all nodes on flow.

Kind regards,

Gauthier

**Reply by Knabbler** [reply]
Thank you.

Kind regards

Knabbler
